examples/ProjPlaneCurve.jl

```diff
@@ -0,0 +1,534 @@
+module ProjPlaneCurveModule
+
+using Oscar, Markdown
+
+include("AffinePlaneCurve.jl")
+using .AffinePlaneCurveModule
+
+import Base.:(==)
+
+export ProjPlaneCurve, degree, jacobi_ideal, issmooth, tangent,
+       curve_components, isirreducible, isreduced, common_component, reduction,
+       dehomogenization, curve_intersect, curve_singular_locus, issmooth_curve
+
+################################################################################
+#
+# We follow the lecture notes by Andreas Gathmann:
+# https://www.mathematik.uni-kl.de/~gathmann/en/curves.php
+#
+################################################################################
+#
+
+mutable struct ProjPlaneCurve{S}
```

Comment:
What is `S`? I am sure it's crystal clear for you, but for me just looking at this code, it's unclear. Perhaps this is the type of the base field? A comment before this line indicating that would be helpful, and/or a type constraint for `S` that makes it clear(er). The same holds for `AffinePlaneCurve`.

From looking elsewhere in the code, I suspect you want this:
```suggestion
mutable struct ProjPlaneCurve{S <: FieldElem}
```

Reply:
Also, I wonder: should affine and projective plane curves perhaps have a common supertype, something like `abstract type PlaneCurve{S} end` ? This might make sense if they have common functionality with similar semantics; then some functions wouldn't have to be written multiple times. E.g. the `hash` and `==` functions seems to be identical for both. 

That doesn't necessarily mean it's a good idea to let them have a shared supertype; so this is an honest, open question.